credentials.go

```diff
@@ -0,0 +1,104 @@
+package gojenkins
+
+import (
+	"bytes"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"net/http"
+	"net/url"
+	"strings"
+)
+
+type SecretCreds struct {
+	Scope       string `json:"scope"`
+	ID          string `json:"id"`
+	Secret      string `json:"secret"`
+	Description string `json:"description"`
+	Class       string `json:"$class"`
+}
+
+type UsernamePasswordCredential struct {
+	Scope       string `json:"scope"`
+	User        string `json:"username"`
+	Password    string `json:"password"`
+	ID          string `json:"id"`
+	Description string `json:"description"`
+	Class       string `json:"$class"`
+}
+
+func (jenkins *Jenkins) CreateCredentialsSecret(scope, id, description, secret string) {
+	header := map[string]string{"Content-Type": "application/x-www-form-urlencoded"}
+
+	m := SecretCreds{
+		Scope:       scope,
+		ID:          id,
+		Secret:      secret,
+		Description: description,
+		Class:       "org.jenkinsci.plugins.plaincredentials.impl.StringCredentialsImpl",
+	}
+
+	b := new(bytes.Buffer)
+	json.NewEncoder(b).Encode(m)
+	jsonPayload := `{"": "0", "credentials":` + b.String() + `}`
+	jenkins.createCredentials(header, jsonPayload)
+}
+
+func (jenkins *Jenkins) postForm(path string, additionalHeader map[string]string, params url.Values, data url.Values, body interface{}) (err error) {
+	requestURL := jenkins.baseUrl + path
+	if params != nil {
+		queryString := params.Encode()
+		if queryString != "" {
+			requestURL = requestURL + "?" + queryString
+		}
+	}
+
+	req, err := http.NewRequest("POST", requestURL, strings.NewReader(data.Encode()))
+
+	for k, v := range additionalHeader {
+		req.Header.Add(k, v)
+	}
+	if _, err := jenkins.checkCrumb(req); err != nil {
+		return err
+	}
+
+	resp, err := jenkins.sendRequest(req)
+	if err != nil {
+		return
+	}
+	if !(200 <= resp.StatusCode && resp.StatusCode <= 299) {
```

Comment:
Can you narrow this condition? In the success case, do we have to expect the 200-299 status code? What about 204 Not Content?

Reply:
In the case of a new secret creation or an existing secret update request, Jenkins returns  200 code only.
https://github.com/yosida95/golang-jenkins/blob/1b8eb2ef03ecce03532d9ae54b2497325b01e666/credentials.go#L68-L69